.github/workflows/check.yml

```
@@ -0,0 +1,15 @@
+on: [push, pull_request]
+name: CI Checks
+jobs:
+  check:
+    name: Check and Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: thumbv6m-none-eabi
+      - run: "cargo check --all --examples"
+      - run: "cargo fmt -- --check"
+      - run: "cargo clippy -- -Dwarnings"
```

Using [`actions-rs/cargo`](https://github.com/actions-rs/cargo) makes the error pretty-printed in the GitHub Actions UI, instead of having to scroll through the logs. Also, the `--all` flag of `cargo check` is deprecated for `--workspace`.

```suggestion
      - uses: actions-rs/cargo@v1
        with:
          command: check
          args: --workspace
      - uses: actions-rs/cargo@v1
        with:
          command: fmt
          args: -- --check
      - uses: actions-rs/cargo@v1
        with:
          command: clippy
          args: -- -Dwarnings
```